astro/src/content/blog/fusionauth-on-fly-io.mdx

```diff
@@ -0,0 +1,360 @@
+      - fusionauth_config:/usr/local/fusionauth/config
+
+networks:
+  search_net:
+    driver: bridge
+
+volumes:
+  fusionauth_config:
+  search_data:
+
+```
+
+
+Start FusionAuth using the docker compose.
+
+```sh
+docker compose up
+```
+
+
+### External Fly.io to Local Connection
+
+<Aside type="note">
+If you are having any issues find more details about [Fly.io Postgres External Connections](https://fly.io/docs/postgres/connecting/connecting-external/).
+</Aside>
+
+To connect across the internet and not just within Fly.io you must allocate an external IP v4.
+
+```sh
+fly ips allocate-v4 --app fusion-fly-db
+```
+
+You will see a message `? Looks like you're accessing a paid feature. Dedicated IPv4 addresses now cost $2/mo. Are you ok with this? (y/N) y`, answer yes to create the new IP address.
+
+Once the external IP address has been allocated you can now connect from your local FusionAuth instance to Fly.io's Postgres database.
+
+Update the `.env` file taking note of the new POSTGRES_URL `fusion-fly-db.fly.dev`.
+
+```
+POSTGRES_URL=jdbc:postgresql://fusion-fly-db.fly.dev:5432/postgres
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=vGrUf0wysPLwQFc
+DATABASE_USERNAME=fusionauth
+DATABASE_PASSWORD=hkaLBM3RVnyYeYeqE3WI1w2e4Avpy0Wd5O3s3
+ES_JAVA_OPTS="-Xms512m -Xmx512m"
+FUSIONAUTH_APP_MEMORY=512M
+```
+
+## Connecting Fly.io FusionAuth Application
+
+Now that you have seen how to test FusionAuth locally, it is time to try using it on Fly.io.
+
+Fly.io does not currently support using `docker compose` so you will need to create a `Dockerfile` using the below command.
+
+```sh
+touch Dockerfile
+```
+
+Update `Dockerfile` with the below information.
+
+```Dockerfile
+# Docker file content 
+FROM  fusionauth/fusionauth-app:latest as build
+FROM debian:buster-slim
+RUN apt-get update && apt-get install -y \
+    curl \
+    ca-certificates \
+    --no-install-recommends
+# copy FusionAuth from other image.
+COPY  --from=build /usr/local/fusionauth /usr/local/fusionauth
+#Addd Env
+ENV FUSIONAUTH_APP_URL="http://0.0.0.0:9011"
+CMD ["/usr/local/fusionauth/fusionauth-app/bin/start.sh"]
+```
+
+Using Fly.io's command line tool you can now launch the FusionAuth application using the above `Dockerfile`.
+
+```sh
+fly launch
+```
+
+You will see the following when you launch, this is where you will include your application name like `fusion-fly`.
+
+```sh
+Scanning source code
+Detected a Dockerfile app
+? Choose an app name (leave blank to generate one):
+```
+
+Then select the region you would like this deployed in, for example `Denver, Colorado (US) (den)`. Pick one that is closest to your users or just you for testing.
+
+You will then see the message `Would you like to set up a Postgresql database now? (y/N)`
+If you followed the steps above answer no, if not answer yes and see the steps from above.
+
+This will create a `fly.toml` file for the app in the current directory that has the information in it like below.
+
+```toml
+app = "fusion-fly"
```

Comment:
Ah, after playing, I had to update this to make sure it is my app name.  Thing this needs to be more clear.